src/actions/ui.js

```diff
@@ -1,12 +1,18 @@
 // @flow
 const constants = require("../constants");
-const { getSource } = require("../selectors");
+const { getSource, getFileSearchState } = require("../selectors");
 import type { ThunkArgs } from "./types";
 
 function toggleFileSearch(searchOn: boolean) {
-  return {
-    type: constants.TOGGLE_FILE_SEARCH,
-    searchOn
+  return ({ dispatch, getState }: ThunkArgs) => {
+    if (searchOn) {
```

Comment:
lol, this could get hairy

for instance `toggleFileSearch(true)` or `toggleFileSearch(false)`

I think I'd just change the API to `toggleFileSearch()` and then always flip it here. KISS keep it simple stupid lol